src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -142,10 +146,16 @@ object Settings {
           missingArg
       }
 
-      if (prefix != "" && arg.startsWith(prefix))
-        doSet(arg drop prefix.length)
-      else if (prefix == "" && name == arg.takeWhile(_ != ':'))
-        doSet(arg.dropWhile(_ != ':').drop(1))
+      if (prefix != "")
```

Comment:
I think we can leave the old logic unchanged here. StringSettings should behave like MultiString settings (those which have a ListTag), that's all.